hail/python/hail/dev-environment-py3.7.yml

```diff
@@ -0,0 +1,20 @@
+name: hail-py3.7
+dependencies:
+- python=3.7
+- Sphinx>=1.8.1
```

Comment:
it's not safe to do `>=` - for instance, sphinx 1.8.5 seems not to work to build Hail with certain versions of pyspark. Can you change these just to `=`?

Reply:
My intention was to have a more flexible environment that updated when dependencies did so it was easier to track and keep dependencies up to date. But I'll pin these.

Reply:
I totally get the reasoning, yeah

Reply:
@chrisvittal @tpoterba IMO, we should be pinning to a major version.

Do we know why Sphinx broke compatibility between two patch versions?

Reply:
no. Chris was trying to investigate that and sphinx removed his file system in retaliation.

Reply:
I made another review with my more general comments about version pinning. I think there's a middle path between patch-pinning and total forward compatibility.